components/tools/OmeroJava/test/integration/thumbnail/SkipThumbnailsPermissionsTest.java

```diff
@@ -0,0 +1,686 @@
+        re = factory.createRenderingEngine();
+        re.lookupPixels(pixelsId);
+        if (!re.lookupRenderingDef(pixelsId)) {
+            re.resetDefaultSettings(true);
+            re.lookupRenderingDef(pixelsId);
+        }
+        re.load();
+        re.setActive(0, false);
+        re.saveCurrentSettings();
+        re.close();
+
+        // Get thumbnail for user 1
+        byte[] user1Thumbnail = Utils.getThumbnailWithoutDefault(svc);
+        Utils.checkSize(user1Thumbnail, 96, 96);
+        svc.close();
+
+        // Check that the thumbnails are different
+        Assert.assertFalse(Arrays.equals(user1Thumbnail, user2Thumbnail));
+    }
+
+    /**
+     * Sets the group permissions and the user's ones.
+     *
+     * @return
+     */
+    @SuppressWarnings("Duplicates")
+    @DataProvider(name = "permissions")
+    public Object[][] providePermissions() {
+        int index = 0;
+        final int PERMISSION = index++;
+        final int IS_ADMIN = index++;
+        final int IS_GROUP_OWNER = index++;
+
+        boolean[] booleanCases = new boolean[]{false, true};
+        String[] permsCases = new String[]{"rw----", "rwr---", "rwra--", "rwrw--"};
+        List<Object[]> testCases = new ArrayList<>();
+
+        for (boolean isAdmin : booleanCases) {
+            for (boolean isGroupOwner : booleanCases) {
+                for (String groupPerms : permsCases) {
+                    Object[] testCase = new Object[index];
+                    testCase[PERMISSION] = groupPerms;
+                    testCase[IS_ADMIN] = isAdmin;
+                    testCase[IS_GROUP_OWNER] = isGroupOwner;
+                    testCases.add(testCase);
+                }
+            }
+        }
+        return testCases.toArray(new Object[testCases.size()][]);
+    }
+
+    /**
+     * Creates a 512x512 fake image.
+     *
+     * @param extension
+     * @return
+     * @throws Throwable
+     */
+    private Pixels importFile(ImportConfig config) throws Throwable {
+        File file = createImageFile("fake");
+        return importFile(config, file, "fake").get(0);
+    }
+
+    /**
+     * Creates a large fake image. A pyramid will be generated.
+     *
+     * @param extension
+     * @return
+     * @throws Throwable
+     */
+    private Pixels importLargeFile(ImportConfig config) throws Throwable {
+        File f = File.createTempFile("bigImageFake&sizeX=3500&sizeY=3500&little=false", ".fake");
+        f.deleteOnExit();
+        return importAndWaitForPyramid(config, f, "fake");
+    }
+
+    /**
+     * Creates a 512x512 fake image.
+     *
+     * @param extension
+     * @return
+     * @throws Throwable
+     */
+    private File createImageFile(String extension) throws Throwable {
+        File f = File.createTempFile("imageFake", "."+ extension);
+        f.deleteOnExit();
+        return f;
+    }
+
+    private void triggerPyramidGeneration(long pixelsId) throws ServerError {
```

Comment:
@joshmoore: I suggested this approach when you were on leave. Absolutely open to less hacky ideas.

Reply:
Unless we extend the methods in `omero/cmd/FS.ice` this is probably find for a test (and is equivalent to loading the image itself)

Reply:
Thanks